src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -40,11 +46,37 @@ type Props = {
   actions: Action[];
   className?: string;
   triggerIndex: number;
-  onAdd: (type: Action['type']) => void;
+  onAdd: (action: Action) => void;
   onChange: (actions: Action[]) => void;
 };
 
+/**
+ * Lists saved actions as well as control to add a new action
+ */
 class ActionsPanel extends React.PureComponent<Props> {
+  /**
+   * Actions have a type (e.g. email, slack, etc), but only some have
+   * an integrationId (e.g. email is null). This helper creates a unique
+   * id based on the type and integrationId so that we know what action
+   * a user's saved action corresponds to.
+   */
+  getActionUniqueKey({type, integrationId}: Pick<Action, 'type' | 'integrationId'>) {
+    return `${type}-${integrationId}`;
```

Comment:
are you sure that we will not have duplicate actions's type?

Reply:
Yeah, the combination of `type` and `integrationId` will be unique, e.g. if you have more than one of the same `type`, you must have an `integrationId` (e.g. type = slack, and id = id of your workspace)

Reply:
I approve of this unique key logic.